### ISSUE TITLE
Get-AzureRmResource -ResourceGroupName myrgname fails

### ISSUE THREAD

**Issue by ayeltsov** [Azure/azure-powershell#3539]
This was supposed to have been fixed - see 1493, but it's still broken

```
C:\Users\ayeltsov> Get-AzureRmResource -ResourceGroupName jobs
Get-AzureRmResource : Parameter set cannot be resolved using the specified named parameters.
At line:1 char:1
+ Get-AzureRmResource -ResourceGroupName jobs
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Get-AzureRmResource], ParameterBindingException
    + FullyQualifiedErrorId : AmbiguousParameterSet,Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.Ge
   tAzureResourceCmdlet

```

```
C:\Users\ayeltsov> get-command Get-AzureRmResource

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Cmdlet          Get-AzureRmResource                                3.6.0      AzureRM.Resources

```

**Comment by markcowl** [comment]
@ravbhatnagar We should fix this.  Please let me know if you have questions.

**Comment by coryallegory** [comment]
`FindAzureRmResource -ResourceGroupNameEquals jobs`

also fails. Current workaround is to make a naked `Get-AzureRmResource` call which returns A LOT of resources, and manually filter clientside.

**Comment by ravbhatnagar** [comment]
@coryallegory - This is the solution for this scenario going forward. We don't plan to add support for getting collections using Get-AzureRmResource. For any querying, please use the Find-AzureRmResource. For getting a single resource, please use Get-AzureRmResource.

**Comment by coryallegory** [comment]
Can we get the documentation updated to reflect that we can't use the ResourceGroupName parameter by itself for Get-AzureRmResource? That would have saved me a great deal of time as the examples are misleading.

https://docs.microsoft.com/en-us/powershell/resourcemanager/azurerm.resources/v2.0.3/get-azurermresource

**Comment by coryallegory** [comment]
Or at least calling `Get-AzureRmResource -?` specifically states that "By default, it gets all resources in the subscription, but you can use the parameters in the cmdlet to filter the results."

**Comment by ravbhatnagar** [comment]
yes, I will have the docs updated. Apologies for this. @tfitzmac @vivsriaus @tianoms - Another example where the docs don't match the functionality. Lets fix this.

**Comment by coryallegory** [comment]
Thanks! Also I am impressed by the speed of responses :+1:

**Comment by cormacpayne** [comment]
This will be resolved in PR https://github.com/Azure/azure-powershell/pull/6008

**Comment by cormacpayne** [comment]
The fix for this should be available on May 7th in version 6.0.0 of `AzureRM` (version 6.0.0 of `AzureRM.Resources`). Please re-open this issue if the problem persists.
